ManageSieve: make sieve_formattime() parse all time-formatters PHP does.

bennet0496 opened the pull request:
Roundcube's `$config['time_format']` understands all PHP time and especially timezone format specifiers listed in the [PHP docs](https://www.php.net/manual/en/datetime.format.php); however, `sieve_formattime()` that used this value via `rcmail.env.time_format` does not.

Therefore if the config value e.g. is `$config['time_format'] = "H:i T"` to include the abbreviated timezone name (which works else where, like the message list), then forms using `sieve_formattime()`, the OOO form, will generate times like "6:00 T" and so on
![image](https://github.com/user-attachments/assets/0f3f9e9e-9c3f-476e-95af-e64d09330894)
This causes the actual time be set to "Tango Time", i.e. UTC-7
![image](https://github.com/user-attachments/assets/9ca57287-6bc6-41b0-841f-76cf3329a7d3)
Which for users outside of UTC-7 is unexpected behavior.

Therefore I changed `sieve_formattime()` to understand any formatter that may reasonably (or unreasonably, like microseconds or Unix Time. But never say never) be configured by an admin or user. For the timezone values, I opted to pass these values from PHP to JS via `set_env()` as in JS these values are extraordinarily hard to reliably to obtain without additional libraries.
![image](https://github.com/user-attachments/assets/88dad7b7-3ec1-4c3e-902f-b30d3d9ae2ed)


I also changed some thing at the beginning of `sieve_formattime()` that looked odd to me. Like per-declaring all variables. And actually used the value that `h` would have been in the `g` and `h` cases. But I'm also happy reverting these changes if they are undesired.

pabzm commented:
Thank you very much for your contribution!

Your solution is solid, but please don't include the drive-by improvements into the same PR (or at least not into the same commit). It makes it unnecessary hard to comprehend, which changes are related.

Could you correct that?

bennet0496 committed:
Managesieve: Parse all time and timezone formatters in sieve_formattime() that PHP understands

bennet0496 commented:
I removed the drive-by changes to the 12h format. Do you want the fix for that here as another commit or do you want another PR for that?

Also I removed the Unix timestamp, because looking over it, it doesn't quite makes sense here, because of the implicitly included date.

pabzm commented:
Please open a new PR for the 12h-changes. Thank you!

pabzm commented:
Great, thanks! Now all that's missing from my point of view is an entry to the Changelog. Would you mind adding one? (Feel free to decline!)

@alecpl Any thoughts on this?

bennet0496 committed:
Update changelog for PR#9655

bennet0496 commented:
Is the entry ok like this? If not I will revert it :)

pabzm commented:
Perfect, thank you! 👍

pabzm commented:
@alecpl I'll merge this after one more week without a comment.

/remind me to merge in one week

pabzm commented:
@bennet0496 Can you rebase to resolve the conflict, please?

pabzm commented:
@bennet0496 Thanks again for your contribution! 🎊
